compose/ui/ui/src/jsWasmMain/kotlin/androidx/compose/ui/window/ComposeWindow.js.kt

```diff
@@ -82,3 +99,59 @@ internal actual class ComposeWindow actual constructor() {
         layer.dispose()
     }
 }
+
+private val defaultCanvasElementId = "ComposeTarget"
+
+@ExperimentalComposeUiApi
+/**
+ * EXPERIMENTAL! Might be deleted or changed in the future!
+ *
+ * Initializes the composition in HTML canvas identified by [canvasElementId].
+ *
+ * It can be resized by providing [requestResize].
+ * By default it will listen to the window resize events.
+ */
+fun CanvasBasedWindow(
+    title: String = "JetpackNativeWindow",
+    canvasElementId: String = defaultCanvasElementId,
+    requestResize: (suspend () -> IntSize)? = null,
+    content: @Composable () -> Unit = { }
+) {
+
+    val resize: suspend () -> IntSize = if (requestResize != null) {
+        requestResize
+    } else {
+        val channel = Channel<IntSize>(capacity = CONFLATED)
+
+        window.addEventListener("resize", { _ ->
+            val w = document.documentElement?.clientWidth ?: 0
+            val h = document.documentElement?.clientHeight ?: 0
+            channel.trySend(IntSize(w, h))
+        })
+
+        suspend {
+            channel.receive()
+        }
```

Comment:
Else branch — it is default resize behavior.
I think we can extract it to separate function:
```
    suspend fun defaultResizeBehavior() -> IntSize {
        val channel = Channel<IntSize>(capacity = CONFLATED)
        window.addEventListener("resize", { _ ->
            val w = document.documentElement?.clientWidth ?: 0
            val h = document.documentElement?.clientHeight ?: 0
            channel.trySend(IntSize(w, h))
        })

        suspend {
            channel.receive()
        }
    }
```